bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Pattern.java

```diff
@@ -39,22 +41,19 @@
  */
 public class Pattern extends Resource {
 
-	/**
-	 * the OS resource for the Pattern
-	 * (Warning: This field is platform dependent)
-	 * <p>
-	 * <b>IMPORTANT:</b> This field is <em>not</em> part of the SWT
-	 * public API. It is marked public only so that it can be shared
-	 * within the packages provided by SWT. It is not available on all
-	 * platforms and should never be accessed from application code.
-	 * </p>
-	 *
-	 * @noreference This field is not intended to be referenced by clients.
-	 */
-	public long handle;
+	private int initialZoom;
 
 	private Runnable bitmapDestructor;
 
+	private Image image;
+
+	private float x1, y1, x2, y2;
+	private Color color1, color2;
+	private int alpha1, alpha2;
+	private final boolean isImagePattern;
```

Comment:
Can you find a better name? One that doesn't need `is...`? The reason is that usually the **getters (methods)** of boolean fields start with `is...` / `has...` / `needs...` etc

Reply:
In that case:
1. Declare `image` as `final`
2. Explicitly set it to `null` in the corresponding constructor
3. Create a private method `isImagePattern()` that returns `image != null`

Reply:
This is also done and can be resolved